RESTserver/src/main/java/com/softserve/if072/restservice/service/security/ProductSecurityService.java

```diff
@@ -21,4 +21,20 @@ public boolean hasPermissionToAccess(int productId){
         Product product = productDAO.getByID(productId);
         return product != null && product.getUser() != null && product.getUser().getId() == getCurrentUser().getId();
     }
+
+    public boolean hasPermissionToAccess(String productName){
+
+        Product product = productDAO.getProductByName(productName);
+
+        if(product == null) {
+            return true;
+        } else {
+            if(product != null && product.getUser() != null && product.getUser().getId() == getCurrentUser().getId()) {
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
```
            if(product != null && product.getUser() != null && product.getUser().getId() == getCurrentUser().getId()) {
                return true;
            } else {
                return false;
            }
```
Can be replaced with
`return product != null && product.getUser() != null && product.getUser().getId() == getCurrentUser().getId()`